middleware/redirect/redirect.go

```diff
@@ -22,7 +23,21 @@ func (rd Redirect) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error
 	for _, rule := range rd.Rules {
 		if rule.From == "/" {
 			// Catchall redirect preserves path (TODO: Standardize/formalize this behavior)
-			newPath := strings.TrimSuffix(rule.To, "/") + r.URL.Path
+			toURL, err := url.ParseRequestURI(rule.To)
+			if err != nil {
+				return 500, err
+			}
+			newPath := toURL.Host + toURL.Path + r.URL.Path
+			rmSlashs := regexp.MustCompile("//+")
+			newPath = rmSlashs.ReplaceAllString(newPath, "/")
```

Comment:
The main problem with path.Join is that the path is Cleaned afterwards. This means that if you have a "/" at the end it will be removed.